### PR TITLE
Small improvements

### PR DESCRIPTION
**Pull request by pblasucci**
- Added `warn` combinator
- Renamed `Combinators` module to `Trial`
- Added several missing XmlDoc comments

### CODE REVIEW

**Comment by forki** [comment]
I'm on vacation, but feel free to :shipit: and release it.

File: src/Chessie/ErrorHandling.fs [hunk 1]
```diff
@@ -3,11 +3,10 @@ namespace Chessie.ErrorHandling
 
 open System
 
- 
 /// Represents the result of a computation.
 type Result<'TSuccess, 'TMessage> = 
     /// Represents the result of a successful computation.
-    | Ok of 'TSuccess * 'TMessage list
+    | Okay of 'TSuccess * 'TMessage list
```
Comment (forki) [inline src/Chessie/ErrorHandling.fs]: Not a big fan of this.

File: src/Chessie/ErrorHandling.fs [hunk 2]
```diff
@@ -18,25 +17,28 @@ type Result<'TSuccess, 'TMessage> =
...
-[<AutoOpen>]
 /// Basic combinators and operators for error handling.
-module Combinators =       
+[<AutoOpen>]
+module Trial =  
     /// Wraps a value in a Success
-    let inline ok<'TSuccess,'TMessage> (x:'TSuccess) : Result<'TSuccess,'TMessage> = Ok(x, [])
+    let inline ok<'TSuccess,'TMessage> (x:'TSuccess) : Result<'TSuccess,'TMessage> = Okay(x, [])
+
+    /// Wraps a value in a Success and adds a message
```
Comment (forki) [inline src/Chessie/ErrorHandling.fs:43]: Could you please use it in a test? Thanks

**Comment by pblasucci** [comment]
After thinking about thing for a bit, I reverted the `Result` union back to `OK` or `Fail`. I also added a `pass` function (basically an alias to `ok`), and an Active Pattern for looking at a result as being either:
- `Pass` ... value and no messages
- `Warn` ... value plus messages
- `Fail` ... no value, just messages

These changes keep the original code basically unchanged. However, they allow for an additional "three-track" workflow which fits _inside_ the "two-track" ROP model. I also added a help doc demonstrating the usage of the new constructs.

File: docs/tools/templates/template.cshtml [hunk 3]
```diff
@@ -46,6 +46,7 @@
             
             <li class="nav-header">F# Articles</li>
             <li><a href="@Root/railway.html">Railway-oriented programming</a></li>
+            <li><a href="@Root/passawrnfail.html">Handling warnings and errors</a></li>
```
Comment (mexx) [inline docs/tools/templates/template.cshtml]: please correct the typo: should be `passwarnfail.html`

File: src/Chessie/ErrorHandling.fs [hunk 4]
```diff
@@ -3,11 +3,10 @@ namespace Chessie.ErrorHandling
 
 open System
 
- 
 /// Represents the result of a computation.
 type Result<'TSuccess, 'TMessage> = 
     /// Represents the result of a successful computation.
-    | Ok of 'TSuccess * 'TMessage list
+    | OK of 'TSuccess * 'TMessage list
```
Comment (mexx) [inline src/Chessie/ErrorHandling.fs]: What does the change of case bring for the user? I wouldn't change it.

File: src/Chessie/ErrorHandling.fs [hunk 5]
```diff
@@ -144,8 +147,20 @@ module Combinators =
...
+    /// Categorizes a result based on its state and the presence of extra messages
+    let inline (|Pass|Warn|Fail|) result =
+      match result with
+      | OK  (value, []  ) -> Pass  value
+      | OK  (value, msgs) -> Warn (value,msgs)
+      | Fail        msgs  -> Fail        msgs
+
+    let inline failOnWarnings result =
+      match result with
+      | Warn (_,msgs) -> Fail [] |> mergeMessages msgs
```
Comment (mexx) [inline src/Chessie/ErrorHandling.fs]: Wouldn't `Warn (_,msgs) -> Fail msgs` bring the same result?

File: src/Chessie/ErrorHandling.fs [hunk 6]
```diff
@@ -18,25 +17,31 @@ type Result<'TSuccess, 'TMessage> =
...
+module Trial =  
+    /// Wraps a value in a Success
+    let inline ok<'TSuccess,'TMessage> (x:'TSuccess) : Result<'TSuccess,'TMessage> = OK(x, [])
+
     /// Wraps a value in a Success
-    let inline ok<'TSuccess,'TMessage> (x:'TSuccess) : Result<'TSuccess,'TMessage> = Ok(x, [])
+    let inline pass<'TSuccess,'TMessage> (x:'TSuccess) : Result<'TSuccess,'TMessage> = OK(x, [])
+
+    /// Wraps a value in a Success and adds a message
+    let inline warn<'TSuccess,'TMessage> (msg:'TMessage) (x:'TSuccess) : Result<'TSuccess,'TMessage> = Result.Succeed (x,msg)
```
Comment (mexx) [inline src/Chessie/ErrorHandling.fs]: I would implement this like other wrapper functions for consistency. `Ok(x, [ msg ])`

**Comment by mexx** [comment]
Honestly I find such PRs really hard to review, as there are multiple things going on and the reformatting adds more struggle.
IMHO Mark Seemann (aka @ploeh) has a great [write](http://blog.ploeh.dk/2015/01/15/10-tips-for-better-pull-requests/) on this.

**Comment by pblasucci** [comment]
So, I don't normally like to merge my one pull requests. But this one's been sitting out a while (and at this point I've incorporated all of the feedback), so I thought I'd go and release it. However, when I looked at the repo a bit more closely, I'm very confused. I see that GitHub Releases and NuGet have version numbers which don't correspond to the [Release Notes](https://github.com/fsprojects/Chessie/blob/master/RELEASE_NOTES.md). 

So, what's up with that?

**Comment by forki** [comment]
Sorry I'm on vacation. Otherwise I'd have merged and released it.

Feel free to shipit. Maybe use 0.1 as version number

**Comment by pblasucci** [comment]
@forki Sorry to bother you on holiday... I'll update the version and ship it.

Thanks!

**Pull request merged**